plugins/producer_plugin/test/test_snapshot_scheduler.cpp

```diff
@@ -118,8 +120,13 @@ BOOST_AUTO_TEST_CASE(snapshot_scheduler_test) {
          // one of the snapshots is done here and request, corresponding to it should be deleted
          BOOST_CHECK_EQUAL(2, pp->get_snapshot_requests().snapshot_requests.size());
 
-         // check whether no pending snapshots present
-         BOOST_CHECK_EQUAL(0, pp->get_snapshot_requests().snapshot_requests.begin()->pending_snapshots->size());
+         // check whether no pending snapshots present for a snapshot with id 0
+         const auto& snapshot_requests = pp->get_snapshot_requests().snapshot_requests;
+         auto it = find_if(snapshot_requests.begin(), snapshot_requests.end(),[](const producer_plugin::snapshot_schedule_information& obj) {return obj.snapshot_request_id == 0;});
+         if (it!=snapshot_requests.end()) {
+            auto& pending = it->pending_snapshots;
+            BOOST_CHECK_EQUAL(0, pending ? pending->size() : 0);
```

Comment:
Is there any situation in which `pending_snapshots` can exist and its value is 0? 

If so, what is the meaning behind such a case and how does it differ from `pending_snapshots` existing but being empty.

If not, this check should be replaced with the stricter check `BOOST_CHECK(!it->pending_snapshots);`.

Reply:
Thank you, updated. pending_snapshots is lazy initialized when first pending snapshot is created and will be returned empty when there is no pending snapshots at the moment. Probably as improvement this can be changed in future to either always return pending_snapshots in request reply or just reset() an optional when no pending snapshot present.